src/Relude/List.hs

```diff
@@ -13,7 +11,39 @@ Utility functions to work with lists.
 module Relude.List
        ( module Relude.List.NonEmpty
        , module Relude.List.Reexport
+       , (!!?)
        ) where
 
+
+import Relude.Base ((<))
+import Relude.Bool (otherwise)
 import Relude.List.NonEmpty
 import Relude.List.Reexport
+import Relude.Monad (Maybe (..))
+import Relude.Numeric (Int, (-))
+
+
+-- $setup
+-- >>> import Relude
+
+{- | Safer version of 'Relude.Unsafe.!!', returns a Maybe.
+get element from list using index value starting from `0`.
+
+>>> [] !!? 0
+Nothing
+
+>>> ["a", "b", "c"] !!? 3
+Nothing
+
+>>> ["a", "b", "c"] !!? 2
+Just "c"
```

Comment:
Btw, nice documentation examples! But let's add one more with negative index and some non-empty list